rust/src/quic/quic.rs

```diff
@@ -0,0 +1,356 @@
+                found = true;
+                index = i;
+                break;
+            }
+        }
+        if found {
+            self.transactions.remove(index);
+        }
+    }
+
+    fn get_tx(&mut self, tx_id: u64) -> Option<&QuicTransaction> {
+        for tx in &mut self.transactions {
+            if tx.tx_id == tx_id + 1 {
+                return Some(tx);
+            }
+        }
+        return None;
+    }
+
+    fn new_tx(&mut self, header: QuicHeader, data: QuicData) -> QuicTransaction {
+        let mut tx = QuicTransaction::new(header, data);
+        self.tx_id += 1;
+        tx.tx_id = self.tx_id;
+        return tx;
+    }
+
+    fn tx_iterator(
+        &mut self, min_tx_id: u64, state: &mut u64,
+    ) -> Option<(&QuicTransaction, u64, bool)> {
+        let mut index = *state as usize;
+        let len = self.transactions.len();
+
+        while index < len {
+            let tx = &self.transactions[index];
+            if tx.tx_id < min_tx_id + 1 {
+                index += 1;
+                continue;
+            }
+            *state = index as u64;
+            return Some((tx, tx.tx_id - 1, (len - index) > 1));
+        }
+
+        return None;
+    }
+
+    fn parse(&mut self, input: &[u8]) -> bool {
+        match QuicHeader::from_bytes(input, DEFAULT_DCID_LEN) {
+            Ok((rest, header)) => match QuicData::from_bytes(rest) {
+                Ok(data) => {
+                    self.version = header.version.clone().into();
+
+                    let transaction = self.new_tx(header, data);
+                    self.transactions.push(transaction);
+
+                    return true;
+                }
+                Err(e) => {
+                    SCLogDebug!("Quic Data Parsing Error: {:?}", e);
+                    return false;
+                }
+            },
+            Err(e) => {
+                SCLogDebug!("Quic Header Parsing Error: {:?}", e);
+                return false;
+            }
+        }
+    }
+}
+
+// C exports.
+export_tx_get_detect_state!(rs_quic_tx_get_detect_state, QuicTransaction);
+export_tx_set_detect_state!(rs_quic_tx_set_detect_state, QuicTransaction);
+
+#[no_mangle]
+pub extern "C" fn rs_quic_state_new(
+    _orig_state: *mut std::os::raw::c_void, _orig_proto: AppProto,
+) -> *mut std::os::raw::c_void {
+    let state = QuicState::new();
+    let boxed = Box::new(state);
+    return unsafe { transmute(boxed) };
+}
+
+#[no_mangle]
+pub extern "C" fn rs_quic_state_free(state: *mut std::os::raw::c_void) {
+    // Just unbox...
+    let _drop: Box<QuicState> = unsafe { transmute(state) };
+}
+
+#[no_mangle]
+pub extern "C" fn rs_quic_state_tx_free(state: *mut std::os::raw::c_void, tx_id: u64) {
+    let state = cast_pointer!(state, QuicState);
+    state.free_tx(tx_id);
+}
+
+#[no_mangle]
+pub extern "C" fn rs_quic_probing_parser(
+    _flow: *const Flow, _direction: u8, input: *const u8, input_len: u32, _rdir: *mut u8,
+) -> AppProto {
+    if input_len < 5 {
```

Comment:
Why 5 ?
Can you name it as a constant ?
Is it possible to have a UDP packet with its payload lesser than 5 ?